apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/NPIOrgDataLookup.java

```diff
@@ -0,0 +1,128 @@
+package gov.cms.bfd.server.war.commons;
+
+import gov.cms.bfd.server.war.NPIDataUtilityApp;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Provides an NPI org lookup */
+public class NPIOrgDataLookup {
+  private static final Logger LOGGER = LoggerFactory.getLogger(NPIOrgDataLookup.class);
+
+  /** A fake npi number used for testing */
+  public static final String FAKE_NPI_NUMBER = "0000000000";
+
+  /** A fake org name display that is associated with the FAKE_NPI_ORG_NAME */
+  public static final String FAKE_NPI_ORG_NAME = "Fake ORG Name";
+
+  /** Hashmap to keep the org names */
+  private final Map<String, String> npiOrgHashMap = new HashMap<>();
+
+  /** A field to return the testing org lookup */
+  private static NPIOrgDataLookup npiOrgLookupForTesting;
+
+  /** A field to return the production org lookup */
+  private static NPIOrgDataLookup npiOrgLookupForProduction;
+
+  /**
+   * Factory method for creating a {@link NPIOrgDataLookup } for testing that includes the fake org
+   * name.
+   *
+   * @return the {@link NPIOrgDataLookup }
+   */
+  public static NPIOrgDataLookup createNpiOrgLookupForTesting() {
+    if (npiOrgLookupForTesting == null) {
+      npiOrgLookupForTesting = new NPIOrgDataLookup(true);
+    }
+
+    return npiOrgLookupForTesting;
+  }
+
+  /**
+   * Factory method for creating a {@link NPIOrgDataLookup } for production that does not include
+   * the fake org name.
+   *
+   * @return the {@link NPIOrgDataLookup }
+   */
+  public static NPIOrgDataLookup createNpiOrgLookupForProduction() {
+    if (npiOrgLookupForProduction == null) {
+      npiOrgLookupForProduction = new NPIOrgDataLookup(false);
+    }
+
+    return npiOrgLookupForProduction;
+  }
+
+  /**
+   * Constructs an {@link NPIOrgDataLookup}
+   *
+   * @param includeFakeNpiOrgName whether to include the fake testing npi org name or not
+   */
+  private NPIOrgDataLookup(boolean includeFakeNpiOrgName) {
+    readNPIOrgDataFile(includeFakeNpiOrgName);
+  }
+
+  /**
+   * Retrieves the Org Data for NPI Display for the given NPI number using the npi file downloaded
+   * during the build.
+   *
+   * @param npiNumber - npiNumber value in claim records
+   * @return the npi org data display string
+   */
+  public Optional<String> retrieveNPIOrgDisplay(Optional<String> npiNumber) {
```

Comment:
This seems kinda wierd; the signature for the method returns an Optional. The caller will expect an Optional and will probably be testing for .isPresent; so returning a null will probably result in NPE. Shouldn't we just be rturning an Optional.empty() if we don't get a hit?